src/mds/Capability.h

```diff
@@ -189,6 +189,13 @@ class Capability : public Counter<Capability> {
       _issued = caps;
       // don't add bits
       _pending &= caps;
+
+      // if the revoking is not totally finished, add the new
+      // revoking caps back, or in clean_revoke_from() we may
+      // miss removing itself from revoking_caps list later.
+      if (was_revoking && revoking()) {
```

Comment:
Is the additional check for `revoking()` required?

Reply:
However, we really need to be sure if the MDS missed accounting in `_revoke`.

Reply:
> Is the additional check for `revoking()` required?
> 
> EDIT: never mind, that ensures there are more caps to revoke.

Yeah, right.

> However, we really need to be sure if the MDS missed accounting in `_revoke`.

Yeah. Since the client really could send the caps update request to MDS even it couldn't revoke all the caps.